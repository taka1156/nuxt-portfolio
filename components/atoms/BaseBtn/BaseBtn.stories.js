import BaseBtn from './BaseBtn.vue';

export default {
  title: 'Atoms/BaseBtn',
  component: BaseBtn,
  argTypes: {
    btnClick: { action: 'clicked' }
  }
};

const Template = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { BaseBtn },
  template: '<base-btn @btn-click="btnClick">Button</base-btn>'
});

export const Default = Template.bind({});
