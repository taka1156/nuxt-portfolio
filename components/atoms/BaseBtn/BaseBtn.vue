<template>
  <button class="base-btn" :class="`base-btn--${useType}`" @click="btnClick">
    <!-- @slot ボタンラベル -->
    <slot />
  </button>
</template>

<script>
export default {
  name: 'BaseBtn',
  props: {
    useType: {
      type: String,
      default: 'none',
      validator: function (value) {
        return ['none', 'card'].indexOf(value) !== -1;
      },
    },
  },
  methods: {
    btnClick() {
      /**
       * ボタンクリックイベント
       * @event btnClick
       * @type {event}
       */
      this.$emit('btn-click');
    },
  },
};
</script>

<style scoped>
/* css reset */
button {
  margin: 0;
  padding: 0;
  border: none;
  outline: none;
  background: transparent;
  cursor: pointer;
}

/* css reset */

.base-btn--card {
  display: block;
  font-size: 1em;
  width: 75%;
  margin: 10px auto;
  padding: 10px;
  background-color: white;
  color: gray;
  border: 1px solid gray;
  border-radius: 5px;
}

.base-btn--card:active,
.base-btn--card:hover,
.base-btn--card:focus {
  background-color: gray;
  color: white;
}
</style>
