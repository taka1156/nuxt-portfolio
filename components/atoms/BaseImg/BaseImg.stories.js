import BaseImg from './BaseImg.vue';
import { dummyImg } from '@/@/__testdata__/testdata.js';

export default {
  title: 'Atoms/BaseImg',
  component: BaseImg,
  argTypes: {
    size: {
      control: {
        type: 'inline-radio',
        options: ['sm', 'lg']
      }
    }
  }
};

const Template1 = (args, { argTypes }) => ({
  props: Object.keys(argTypes),
  components: { BaseImg },
  template: '<base-img v-bind="$props" />'
});

export const Default = Template1.bind({});
Default.args = dummyImg;
