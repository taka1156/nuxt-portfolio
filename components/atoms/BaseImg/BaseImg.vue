<template>
  <img
    :src="imgUrl"
    :alt="imgAlt"
    class="base-img--extend"
    :class="`base-img--${size}`"
  />
</template>

<script>
export default {
  name: 'BaseImg',
  props: {
    /**
     * 画像のURL
     */
    imgUrl: {
      type: String,
      required: true
    },
    /**
     * 画像が表示できない時に使用する代替テキスト
     * (alt)
     */
    imgAlt: {
      type: String,
      required: true
    },
    /**
     * 画像サイズ
     * @values sm, lg
     */
    size: {
      type: String,
      default: 'sm',
      required: true,
      validator: function (value) {
        return ['sm', 'lg', 'free'].indexOf(value) !== -1;
      }
    }
  }
};
</script>

<style scoped>
img {
  display: block;
  padding: 3px;
}

.base-img--sm {
  width: 20px;
  height: 20px;
}

.base-img--lg {
  width: 50px;
  height: 50px;
}

.base-img--free {
  height: auto;
}
</style>
