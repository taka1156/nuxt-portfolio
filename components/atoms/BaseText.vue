<template functional>
  <p class="base-text" :class="data.staticClass || ''">
    <slot />
  </p>
</template>

<style scoped>
/* css reset */
p {
  margin: 0;
  padding: 0;
}

/* css reset */

.base-text {
  padding: 5px;
}

.base-text--card {
  display: block;
  height: 100px;
  width: 100%;
  margin: 0 auto;
  text-align: left;
  word-wrap: break-word;
  box-sizing: border-box;
}

.base-text--nav {
  display: inline;
  color: white;
}
</style>
