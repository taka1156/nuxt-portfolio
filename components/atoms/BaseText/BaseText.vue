<template>
  <p class="base-text--extend">
    <!-- @slot テキスト -->
    <slot />
  </p>
</template>

<script>
export default {
  name: 'BaseText'
};
</script>

<style scoped>
/* css reset */
p {
  padding: 0;
  margin: 0;
}

/* css reset */
</style>
