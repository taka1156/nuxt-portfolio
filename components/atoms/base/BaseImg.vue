<template>
  <div>
    <img :src="imgUrl" :alt="imgAlt" class="base-img" :class="`base-img--${size}`" />
  </div>
</template>

<script>
export default {
  name: 'BaseImg',
  props: {
    /**
     * 画像のURL
     */
    imgUrl: {
      type: String,
      required: true,
    },
    /**
     * 画像が表示できない時に使用する代替テキスト
     * (alt)
     */
    imgAlt: {
      type: String,
      required: true,
    },
    /**
     * 画像サイズ
     * @values sm, lg
     */
    size: {
      type: String,
      default: 'sm',
      required: true,
      validator: function (value) {
        return ['sm', 'lg'].indexOf(value) !== -1;
      },
    },
  },
};
</script>

<style scoped>
.base-img {
  display: block;
  padding: 3px;
}
.base-img--lg {
  height: 50px;
  width: 50px;
}

.base-img--sm {
  height: 20px;
  width: 20px;
}
</style>
