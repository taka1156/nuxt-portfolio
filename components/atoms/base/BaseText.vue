<template>
  <p class="base-text" :class="`base-text--${useType}`">
    <!-- @slot テキスト -->
    <slot />
  </p>
</template>

<script>
export default {
  name: 'BaseText',
  props: {
    useType: {
      type: String,
      default: 'none',
      validator: function (value) {
        return ['none', 'nav', 'card'].indexOf(value) !== -1;
      },
    },
  },
};
</script>

<style scoped>
/* css reset */
p {
  margin: 0;
  padding: 0;
}

/* css reset */

.base-text {
  padding: 5px;
}

.base-text--card {
  display: block;
  height: 100px;
  width: 100%;
  margin: 0 auto;
  text-align: left;
  word-wrap: break-word;
  box-sizing: border-box;
}

.base-text--nav {
  display: inline;
  color: white;
}
</style>
