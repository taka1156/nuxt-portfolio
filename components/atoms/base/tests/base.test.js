import { shallowMount } from '@vue/test-utils';
import { dummyImg } from '@/__testdata__/testdata.js';
import BaseText from '../BaseText.vue';
import BaseImg from '../BaseImg.vue';
import BaseBtn from '../BaseBtn.vue';

// BaseBtn
describe('BaseBtn', () => {
  const dummyButtonText = 'ダミーボタンテキスト';
  const baseBtn = shallowMount(BaseBtn, {
    slots: {
      default: dummyButtonText,
    },
  });

  it('slotがDOMに反映されているか', () => {
    const btnTag = baseBtn.find('button');
    expect(btnTag.text()).toBe(dummyButtonText);
    expect(btnTag.html()).toMatchSnapshot();
  });

  it('btnClickを実行した時、`btn-click`のイベントが発生するか', () => {
    baseBtn.vm.btnClick();
    expect(baseBtn.emitted('btn-click')).toBeTruthy();
  });
});

// BaseImg
describe('BaseImg', () => {
  const baseImg = propsData =>
    shallowMount(BaseImg, {
      propsData: {
        ...propsData,
        imgUrl: dummyImg.imgUrl,
        imgAlt: dummyImg.imgAlt,
      },
    });

  it('値がDOMに反映されているか(size: sm)', () => {
    const wrapper = baseImg({ size: 'sm' });
    const imgTag = wrapper.find('img');
    expect(imgTag.attributes().src).toBe(dummyImg.imgUrl);
    expect(imgTag.attributes().alt).toBe(dummyImg.imgAlt);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('値がDOMに反映されているか(size: lg)', () => {
    const wrapper = baseImg({ size: 'lg' });
    const imgTag = wrapper.find('img');
    expect(imgTag.attributes().src).toBe(dummyImg.imgUrl);
    expect(imgTag.attributes().alt).toBe(dummyImg.imgAlt);
    // スナップショット
    expect(wrapper.html()).toMatchSnapshot();
  });
});

// BaseText
describe('BaseText', () => {
  const dummyText = 'ダミーテキスト';
  const baseText = propsData =>
    shallowMount(BaseText, {
      slots: {
        default: dummyText,
      },
      propsData: {
        ...propsData,
      },
    });

  it('値がDOMに反映されているか(useType: none)', () => {
    const wrapper = baseText({ useType: 'none' });
    expect(wrapper.text()).toBe(dummyText);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('値がDOMに反映されているか(useType: nav)', () => {
    const wrapper = baseText({ useType: 'nav' });
    expect(wrapper.text()).toBe(dummyText);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('値がDOMに反映されているか(useType: card)', () => {
    const wrapper = baseText({ useType: 'card' });
    expect(wrapper.text()).toBe(dummyText);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
