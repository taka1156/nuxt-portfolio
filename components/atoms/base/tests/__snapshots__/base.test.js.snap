// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseBtn slotがDOMに反映されているか 1`] = `"<button class=\\"btn\\">ダミーボタンテキスト</button>"`;

exports[`BaseText 値がDOMに反映されているか 1`] = `"<p class=\\"base-text\\">ダミーテキスト</p>"`;
